### ISSUE TITLE
FromJava macros fail when a field is named "source"

### ISSUE THREAD

**Issue by alanbaumgartner** [mullvad/jnix#50]
Because the derive proc macros uses the name `source` in its `from_java` implementation, if a field named `source` is not the last field in the struct, the original `source` JObject gets shadowed and fails to parse the rest.

Example
```rust
use jnix::{FromJava, IntoJava};

#[derive(Debug, FromJava, IntoJava)]
#[jnix(package = "x.y.z")]
pub struct Point {
    pub x: i32,
    pub y: i32,
}

#[derive(Debug, FromJava, IntoJava)]
#[jnix(package = "x.y.z")]
pub struct Transport {
    pub source: Point,
    pub destination: Point,
}
```
Error
```
error[E0277]: the trait bound `JObject<'_>: From<Point>` is not satisfied
   --> src\lib.rs:10:17
    |
10  | #[derive(Debug, FromJava, IntoJava)]
    |                 ^^^^^^^^ the trait `From<Point>` is not implemented for `JObject<'_>`
    |
    = help: the following other types implement trait `From<T>`:
              <JObject<'a> as From<&'a AutoLocal<'a, '_>>>
              <JObject<'a> as From<&'a GlobalRef>>
              <JObject<'a> as From<*mut _jobject>>
              <JObject<'a> as From<JByteBuffer<'a>>>
              <JObject<'a> as From<JClass<'a>>>
              <JObject<'a> as From<JList<'a, 'b>>>
              <JObject<'a> as From<JMap<'a, 'b>>>
              <JObject<'a> as From<JString<'a>>>
              <JObject<'a> as From<JThrowable<'a>>>
    = note: required for `Point` to implement `Into<JObject<'_>>`
```

I can write a PR for a fix but I'd like to get some input on how to handle it, a few possible implementations I've thought of are to rename the `source` argument in `from_java` to something that "should" be unused, something like `_source_` perhaps.

Generated code
```rust
impl<'env, 'sub_env> jnix::FromJava<'env, jnix::jni::objects::JObject<'sub_env>> for Transport
    where
        'env: 'sub_env,
{
    const JNI_SIGNATURE: &'static str = concat!("L", "x/y/z/Transport", ";");
    fn from_java(
        env: &jnix::JnixEnv<'env>,
        _source_: jnix::jni::objects::JObject<'sub_env>,
    ) -> Self {
        let class = env.get_class("x/y/z/Transport");
        let source = {
            ...
        };
        let destination = {
            ...
        };
        Self {
            source,
            destination,
        }
    }
}
```

Another possible implementation would be to not have the field names used as the variable being read from java so the generated code would be something like this.

```rust
impl<'env, 'sub_env> jnix::FromJava<'env, jnix::jni::objects::JObject<'sub_env>> for Transport
    where
        'env: 'sub_env,
{
    const JNI_SIGNATURE: &'static str = concat!("L", "x/y/z/Transport", ";");
    fn from_java(
        env: &jnix::JnixEnv<'env>,
        source: jnix::jni::objects::JObject<'sub_env>,
    ) -> Self {
        let class = env.get_class("x/y/z/Transport");
        let field1 = {
            ...
        };
        let field2 = {
            ...
        };
        Self {
            source: field1,
            destination: field2,
        }
    }
}
```

The first suggestion is much easier to implement as its just renaming `source` -> `_source_` but I'm open to here any thoughts

**Comment by dlon** [comment]
I think something like the second suggestion sounds good, or perhaps adding a prefix to the variables names to prevent them from being shadowed.

Renaming `source` is probably an okay workaround for now, but I'd worry about other variables being shadowed in the same way. This is a nitpick, but I also think renaming it to something more normal than `_source_`, but still "unique", like `jnix_source_object`, would be preferable.

**Comment by dlon** [comment]
@alanbaumgartner Since your first suggestion is definitely an improvement, let's just go with that for now.

**Comment by alanbaumgartner** [comment]
@dlon Thanks for the suggestion, Ill get the PR updated 👍

**Comment by dlon** [comment]
Fixed due to #51.
